docs/framework/server-side-editor-api.md

```diff
@@ -0,0 +1,264 @@
+editor.execute( 'replaceAll', 'completely', 'entirely' );
+```
+
+This one line will:
+- Find all instances of "entirely" in your document
+- Change each one to "completely"
+- Keep all the formatting around the changed text
+
+This is perfect for fixing typos, updating old terms, or making any bulk changes to your documents.
+
+### Updating Links in Your Document
+
+Now, consider a more complex scenario. You need to update all links in your document from `/docs/` to `/documents/`. This is a common task when moving content between environments or updating your site structure.
+
+```js
+// Get the root element and create a range that covers all content.
+const root = editor.model.document.getRoot();
+const range = editor.model.createRangeIn( root );
+const items = Array.from( range.getItems() );
+
+// Use model.change to ensure proper undo/redo support.
+editor.model.change( writer => {
+    for ( const item of items ) {
+        let href = item.getAttribute( 'linkHref' );
+
+        if ( item.is( 'textProxy' ) && href ) {
+            // Update the link URL.
+            href = href.replace( '/docs/', '/documents/' );
+            writer.setAttribute( 'linkHref', href, item );
+        }
+    }
+} );
+```
+
+This approach is particularly useful when you need to:
+- Update many links at once
+- Fix broken links across your content
+- Change link patterns in bulk
+
+### Adding Complex HTML Content
+
+Let's say you need to add a pre-made section to your document. This is useful when you want to add templates, import content from other systems, or create dynamic content.
+
+```js
+// The HTML content we want to add.
+const html = '<h2>New section</h2><p>This is a <strong>new section</strong> inserted into the document using <u>server-side editor API</u>.</p>';
+
+// Convert HTML to the editor's model.
+const model = editor.data.parse( html );
+
+// Get the root element and create an insertion position.
+const root = editor.model.document.getRoot();
+const insertPosition = editor.model.createPositionAt( root, 1 );
+
+// Insert the content at the specified position.
+editor.model.insertContent( model, insertPosition );
+```
+
+This is perfect for:
+- Adding content from other places
+- Inserting ready-made templates
+- Adding pre-formatted HTML content 
+
+## Working with Track Changes
+
+{@link features/track-changes Track changes} helps you manage content suggestions effectively, for tasks like automating content review, and implementing AI-powered suggestions.
+
+### Text Modifications
+
+Let's start with a basic text replacement:
+
+```js
+// Enable track changes to mark our edits as suggestions.
+editor.execute( 'trackChanges' );
+
+// Make a simple text replacement.
+editor.execute( 'replaceAll', 'I', 'you' );
+```
+
+The `trackChanges` command ensures that all changes are marked as suggestions, making it easy for others to review and accept or reject them.
+
+### Content Structure Changes
```

Comment:
I wouldn't call it content structure changes. Content structure sounds more like splitting content into blocks, adding headings, using lists and tables, etc.

Reply:
I proposed a new title, which is "Content changes".